src/mame/sinclair/spec_snqk.cpp

```diff
@@ -1766,6 +1775,358 @@ void spectrum_state::snx_decompress_block(address_space &space, uint8_t *source,
 	}
 }
 
+static u16 hrust_decompress_block(uint8_t *dest, uint8_t *source, uint16_t size)
+{
+	class bb_stream
+	{
+	private:
+		u8 *base;
+		u8 *p;
+		int idx;
+		int len;
+		bool eof;
+		u16 bits;
+
+	public:
+		bb_stream(u8 *from, int block_size)
+		{
+			base = p = from;
+
+			len = block_size;
+			idx = 0;
+			eof = false;
+
+			bits = get_byte();
+			bits += 256 * get_byte();
+		}
+
+		u8 get_byte(void)
+		{
+			if (p - base == len)
+			{
+				eof = true;
+				return 0;
+			}
+			return *p++;
+		}
+
+		u8 get_bit()
+		{
+			u16 mask[] = {0x8000, 0x4000, 0x2000, 0x1000, 0x800, 0x400, 0x200, 0x100, 0x80, 0x40, 0x20, 0x10, 0x8, 0x4, 0x2, 0x1};
+
+			u8 bit = (bits & mask[idx]) ? 1 : 0;
```

Comment:
`mask` should be `constexpr`, but do you actually need it?  Can’t you just express this as `u8 bit = BIT(bits, 15 - idx);` anyway?